lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +34,50 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self._path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
 
+        articles_raw = self._path_to_raw_txt_data.glob('*_raw.txt')
+        #meta_info = list(self._path_to_raw_txt_data.glob('*_meta.json'))
+
+        if not self._path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self._path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if not any(self._path_to_raw_txt_data.iterdir()):
+            raise EmptyDirectoryError
+
+        raw_list = [file for file in articles_raw]
```

Comment:
1. you can turn generator into list with just `list`
2. why do you have to turn it into list?